.github/actions/mint-token/action.yml

```diff
@@ -32,16 +35,29 @@ runs:
         id_token_audience: '${{ inputs.service_audience }}'
         id_token_include_email: true
     - id: 'mint-token'
-      shell: "bash"
       env:
-        token: ${{ steps.auth.outputs.id_token }}
-      run: |
-        OIDC_TOKEN=$(curl -sLS "${ACTIONS_ID_TOKEN_REQUEST_URL}&audience=github-token-minter" -H "User-Agent: actions/oidc-client" -H "Authorization: Bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN");
-        JWT=$(echo $OIDC_TOKEN | jq -j '.value');
-        APP_TOKEN=$(curl -X POST -d'{"repositories":["github-token-minter"],"permissions":{"issues":"read"}}' \
-          -H "Authorization: Bearer ${{env.token}}" \
-          -H "X-GitHub-OIDC-Token: ${JWT}" \
-          ${{ inputs.service_url}}/token);
-        TOKEN=$(echo ${APP_TOKEN} | jq -j '.token');
-        echo "token=$(echo $TOKEN)" >> $GITHUB_OUTPUT;
-
+        TOKEN: '${{ steps.auth.outputs.id_token }}'
+      uses: actions/github-script@v6
+      with:
+        script: |
+          try {
+              const {TOKEN} = process.env;
+              const service_url = core.getInput('service_url', { required: true });
+              const request_body = core.getInput('requested_permissions', { required: true});
+              const id_token = await core.getIDToken('github-token-minter');
+              const id_token_value = JSON.parse(id_token).value;
+              const response = await fetch(`${service_url}/token`, {
+                  method: 'post',
+                  body: request_body,
+                  headers: {
+                      'Content-Type': 'application/json',
+                      'X-GitHub-OIDC-Token': id_token_value,
+                      'Authorization': `Bearer ${TOKEN}`,
+                  }
+              });
+              const resp = await response.json();
+              core.setOutput('token', resp.token);
```

Comment:
If this call fails, will we get a NPE on `resp` or is it guaranteed to have a response? If the former, we should check the resp variable for falsey. If the latter, ignore me.

Reply:
I believe for fetch, if an HTTP error code is received, it will fall to the catch block.